config/dynamic-config-builder/src/validator_config.rs

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Error;
+use anyhow::{ensure, Result};
+use libra_config::{
+    config::{ConsensusPeersConfig, NodeConfig, SeedPeersConfig},
+    generator,
+};
+use libra_crypto::{ed25519::Ed25519PrivateKey, PrivateKey, Uniform};
+use libra_types::transaction::Transaction;
+use parity_multiaddr::Multiaddr;
+use rand::{rngs::StdRng, Rng, SeedableRng};
+use std::collections::HashMap;
+use vm_genesis;
+
+const DEFAULT_SEED: [u8; 32] = [13u8; 32];
+const DEFAULT_IPV4_ADVERTISED: &str = "/ip4/127.0.0.1/tcp/6180";
+const DEFAULT_IPV4_LISTEN: &str = "/ip4/0.0.0.0/tcp/6180";
+const DEFAULT_IPV6_ADVERTISED: &str = "/ip6/::1/tcp/6180";
+const DEFAULT_IPV6_LISTEN: &str = "/ip6/::/tcp/6180";
+
+pub struct ValidatorConfig {
+    advertised: Multiaddr,
+    bootstrap: Multiaddr,
```

Comment:
why not having these fields public since we have setters for them?

(Also it'd be nice to have better variable names (`advertised_address` for example) and documentation for the different fields.)

Reply:
no objection to making them public, but reckoned there wasn't much point in doing so

Reply:
will update names to match